mantle/cmd/kola/testiso.go

```diff
@@ -679,55 +679,104 @@ func awaitCompletion(ctx context.Context, inst *platform.QemuInstance, outdir st
-					// better error
-					time.Sleep(1 * time.Second)
-					errchan <- fmt.Errorf("Got EOF from completion channel, %s expected", exp)
-				} else {
-					errchan <- errors.Wrapf(err, "reading from completion channel")
+				errchan <- errors.Wrapf(err, "QEMU unexpectedly exited while awaiting completion")
+			}
+			time.Sleep(1 * time.Minute)
+			errchan <- fmt.Errorf("QEMU exited; timed out waiting for completion")
+		}()
+		go func() {
+			r := bufio.NewReader(qchan)
+			for _, exp := range expected {
+				l, err := r.ReadString('\n')
+				if err != nil {
+					if err == io.EOF {
+						// this may be from QEMU getting killed or exiting; wait a bit
+						// to give a chance for .Wait() above to feed the channel with a
+						// better error
+						time.Sleep(1 * time.Second)
+						errchan <- fmt.Errorf("Got EOF from completion channel, %s expected", exp)
+					} else {
+						errchan <- errors.Wrapf(err, "reading from completion channel")
+					}
+					return
 				}
-				return
+				line := strings.TrimSpace(l)
+				if line != exp {
+					errchan <- fmt.Errorf("Unexpected string from completion channel: %s expected: %s", line, exp)
+					return
+				}
+				plog.Debugf("Matched expected message %s", exp)
 			}
-			line := strings.TrimSpace(l)
-			if line != exp {
-				errchan <- fmt.Errorf("Unexpected string from completion channel: %s expected: %s", line, exp)
-				return
+			plog.Debugf("Matched all expected messages")
+			// OK!
+			errchan <- nil
+		}()
+		go func() {
+			//check for error when switching boot order
+			if booterrchan != nil {
+				if err := <-booterrchan; err != nil {
+					errchan <- err
+				}
 			}
-			plog.Debugf("Matched expected message %s", exp)
-		}
-		plog.Debugf("Matched all expected messages")
-		// OK!
-		errchan <- nil
-	}()
-	go func() {
-		//check for error when switching boot order
-		if booterrchan != nil {
-			if err := <-booterrchan; err != nil {
-				errchan <- err
+		}()
+		err := <-errchan
+		elapsed := time.Since(start)
+		if err == nil {
+			// No error so far, check the console and journal files
+			consoleFile := filepath.Join(outdir, "console.txt")
+			journalFile := filepath.Join(outdir, "journal.txt")
+			files := []string{consoleFile, journalFile}
+			for _, file := range files {
+				fileName := filepath.Base(file)
+				// Check if the file exists
+				_, err := os.Stat(file)
+				if os.IsNotExist(err) {
+					fmt.Printf("The file: %v does not exist\n", fileName)
+					continue
+				} else if err != nil {
+					fmt.Println(err)
+					errchan <- err
+					return elapsed, err
+				}
+				// Read the contents of the file
+				fileContent, err := os.ReadFile(file)
+				if err != nil {
+					fmt.Println(err)
+					errchan <- err
+					return elapsed, err
+				}
+				// Check for badness with CheckConsole
+				warnOnly, badlines := kola.CheckConsole([]byte(fileContent), nil)
+				if len(badlines) == 0 {
+					fmt.Printf("No badness detected in the file: %v\n", fileName)
```

Comment:
probably should remove this log message in the final version of the code

then we can switch this to `if len(badlines) > 0 {`

Reply:
That makes sense, implemented.